test/test_analyze.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright (c) 2025 National Land Survey of Finland (Maanmittauslaitos)
+#
+#  This file is part of geogen-algorithms.
+#
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+import pytest
+from geopandas import GeoDataFrame
+from shapely.geometry import Polygon
+
+from geogenalg.analyze import calculate_coverage
+
+
+def test_calculate_coverage():
+    calculate_coverage(
+        overlay_features=GeoDataFrame(
+            {
+                "id": [1],
+            },
+            geometry=[
+                Polygon([(0, 0), (4, 0), (4, 4), (0, 4), (0, 0)]),
+            ],
+            crs="EPSG:3067",
+        ),
+        base_features=GeoDataFrame(
+            {
+                "class": ["a"],
+                "id": [1],
+            },
+            geometry=[
+                Polygon([(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)]),
+            ],
+            crs="EPSG:3067",
+        ),
+    )
+
+
+def test_calculate_coverage_raises_for_wrong_crs_type():
+    with pytest.raises(ValueError, match="projected CRS"):
+        calculate_coverage(
+            overlay_features=GeoDataFrame(
+                {
+                    "id": [1],
+                },
+                geometry=[
+                    Polygon([(0, 0), (1, 1), (0, 1), (0, 0)]),
+                ],
+            ),
+            base_features=GeoDataFrame(
+                {
+                    "class": ["a"],
+                    "id": [1],
+                },
+                geometry=[
+                    Polygon([(0, 0), (1, 0), (1, 1), (0, 1), (0, 0)]),
+                ],
+            ),
+        )
+
+
+# kolmantena testi tyhjälle dataframelle
+def test_calculate_coverage_empty_input():
+    overlay = GeoDataFrame({"id": []}, geometry=[], crs="EPSG:3067")
+    base = GeoDataFrame({"class": [], "id": []}, geometry=[], crs="EPSG:3067")
+
+    result = calculate_coverage(overlay, base)
+
+    assert result.empty
```

Comment:
You could also add assert for checking that "coverage" is added to the result (if that's how the function is supposed to function?)